### PR TITLE
Fix cmina state transition on reset to factory

### PR DESCRIPTION
**Pull request by olbjo**
Fixes problem with test case AR-ASE.
Adds log error strings for a few errors to sample application.

I have not yet been able to rerun tests on latest with submodule/cmake change

### CODE REVIEW

File: sample_app/sampleapp_common.c [hunk 1]
```diff
@@ -717,13 +717,19 @@ static int app_state_ind (
...
                {
                case PNET_ERROR_CODE_2_ABORT_AR_CONSUMER_DHT_EXPIRED:
-                  error_description = "AR_CONSUMER_DHT_EXPIRED";
+                  error_description = "ABORT_AR_CONSUMER_DHT_EXPIRED";
                   break;
                case PNET_ERROR_CODE_2_ABORT_AR_CMI_TIMEOUT:
                   error_description = "ABORT_AR_CMI_TIMEOUT";
                   break;
                case PNET_ERROR_CODE_2_ABORT_AR_RELEASE_IND_RECEIVED:
-                  error_description = "Controller sent release request.";
```
Comment (pyhys) [inline sample_app/sampleapp_common.c:726]: I prefer human readable descriptive strings for all error cases where we understand what they mean.

File: sample_app/sampleapp_common.c [hunk 2]
```diff
@@ -717,13 +717,19 @@ static int app_state_ind (
...
+                  error_description = "ABORT_AR_RELEASE_IND_RECEIVED";
+                  break;
+               case PNET_ERROR_CODE_2_ABORT_DCP_STATION_NAME_CHANGED:
+                  error_description = "ABORT_DCP_STATION_NAME_CHANGED";
```
Comment (pyhys) [inline sample_app/sampleapp_common.c:729]: "PLC terminated the connection as the device station name changed"

File: sample_app/sampleapp_common.c [hunk 3]
```diff
@@ -717,13 +717,19 @@ static int app_state_ind (
...
+                  break;
+               case PNET_ERROR_CODE_2_ABORT_DCP_RESET_TO_FACTORY:
+                  error_description = "ABORT_DCP_RESET_TO_FACTORY";
```
Comment (pyhys) [inline sample_app/sampleapp_common.c:732]: "Connection was terminated as the PLC did request reset to factory"

File: sample_app/sampleapp_common.c [hunk 4]
```diff
@@ -717,13 +717,19 @@ static int app_state_ind (
                switch (err_code)
                {
                case PNET_ERROR_CODE_2_ABORT_AR_CONSUMER_DHT_EXPIRED:
-                  error_description = "AR_CONSUMER_DHT_EXPIRED";
+                  error_description = "ABORT_AR_CONSUMER_DHT_EXPIRED";
```
Comment (pyhys) [inline sample_app/sampleapp_common.c:720]: "PLC terminated the connection due to missed cyclic data deadline"

**Review by pyhys — APPROVED** [review]
Great! Update the error messages (just a suggestion)

File: sample_app/sampleapp_common.c (hunk 1 above)
Comment (olbjo) [reply sample_app/sampleapp_common.c:726]: I have updated all descriptions.

File: sample_app/sampleapp_common.c (hunk 2 above)
Comment (olbjo) [reply sample_app/sampleapp_common.c]: done, I have updated all descriptions

File: sample_app/sampleapp_common.c (hunk 3 above)
Comment (olbjo) [reply sample_app/sampleapp_common.c]: done, I have updated all descriptions

File: sample_app/sampleapp_common.c (hunk 4 above)
Comment (olbjo) [reply sample_app/sampleapp_common.c]: done, I have updated all descriptions

**Comment by pyhys** [comment]
Sweet!

**Pull request merged**